plugins/org.jboss.tools.openshift.egit.core/src/org/jboss/tools/openshift/egit/core/EGitUtils.java

```diff
@@ -229,7 +229,11 @@ public static Repository createRepository(IProject project,
 	}
 
 	public static File getRepositoryPathFor(IProject project) {
-		return new File(project.getLocationURI().getPath(), Constants.DOT_GIT);
+		File f = new File(project.getLocationURI().getPath());
```

Comment:
Somehow the code you linked looks *way* more complicated... 

The real question is whether my patch is the way the method is intended to behave. If the two are really clones of each other, one should simply call the other, to keep logic isolated and convenience methods obvious.